### PR TITLE
Qa update

### PR DESCRIPTION
**Pull request by geramirez**
Multiple issues from: 18F/foia-hub/issues/538
- [x] fix the emails for the FLRA Dallas and Denver regional offices
- [x] update the FERC's submission link
- [x] remove duplicate emails
- [x] fix the JLP office url 

Also found that the email dups originate from the csv update [here](https://github.com/18F/foia/compare/missing_data?expand=1#diff-08719b9d357973b49df3d5a0eca9b445R88)

Additional updates
- [x] A number of abbreviations disappeared because it seems they were added manually in the last iteration of `layer_with_usa_contacts.py` 
- [x] Reworked the mapping to add a couple more descriptions from USA contacts.

### CODE REVIEW

File: contacts/data/DOI.yaml [hunk 1]
```diff
@@ -355,10 +357,8 @@ departments:
     phone:
     - 703-787-1315
   reading_rooms:
-  - - Deepwater Horizon Library and Reading Room
-    - http://www.boem.gov/Deepwater-Reading-Room/
-  - - Library and Reading Room
```
Comment (ramirezg) [inline contacts/data/DOI.yaml]: Looks like some duplicate reading rooms were removed.

File: contacts/layer_with_usa_contacts.py [hunk 2]
```diff
@@ -111,8 +110,9 @@ def update_dict(old_data, new_data):
 
     old_data['usa_id'] = new_data['usa_id']
     if new_data.get('description'):
-        old_data['description'] = new_data['description']
-    if new_data.get('abbreviation'):
+        old_data['description'] = new_data.get('description')
+    if new_data.get('abbreviation') and not old_data.get('abbreviation'):
+        print(old_data.get('abbreviation'), new_data['abbreviation'])
```
Comment (khandelwal) [inline contacts/layer_with_usa_contacts.py]: I think this print statement might need to be removed?

File: contacts/layer_with_csv.py [hunk 3]
```diff
@@ -85,9 +85,11 @@ def add_contact_info(contacts, row):
     address = organize_address(row)
     if address and 'address' not in office_struct:
         office_struct['address'] = address
-    if row['Email Address']:
-        office_struct['emails'].append(
-            row['Email Address'].replace('mailto:', ''))
+    email = row.get('Email Address')
```
Comment (khandelwal) [inline contacts/layer_with_csv.py:88]: Can you explain this change? What's going on here?

File: contacts/layer_with_usa_contacts.py [hunk 4]
```diff
@@ -134,15 +134,12 @@ def patch_yamls(data, directory):
         with open(filename) as f:
             agency = yaml.load(f.read())
         agency_name = clean_name(agency.get('name'))
-        if not agency.get('description'):
-            if agency_name in data:
-                agency = update_dict(agency, data[agency_name])
-                del data[agency_name]
+        if agency_name in data:
```
Comment (khandelwal) [inline contacts/layer_with_usa_contacts.py:136]: Can you also explain the change here?

**Comment by khandelwal** [comment]
The changes you've made have no explanation with them (it's hard to figure out why they are being made). Another way to do this would also be to provide some tests to indicate what situation is being fixed here.

File: contacts/layer_with_csv.py (hunk 3 above)
Comment (geramirez) [inline contacts/layer_with_csv.py:88]: This change prevents the duplicate email addresses. Originally, emails were appended without checking to make sure the new email wasn't a duplicate. 

I added a test here. 
https://github.com/18F/foia/pull/105/files#diff-af5c68b14c0da4acbb5fe3d2c38d2ef0R156

File: contacts/layer_with_usa_contacts.py (hunk 4 above)
Comment (geramirez) [inline contacts/layer_with_usa_contacts.py:136]: I altered this portion of the script to give priority to the descriptions from the USA contacts API.

File: contacts/layer_with_usa_contacts.py [hunk 5]
```diff
@@ -32,7 +31,7 @@
         'U.S. National Central Bureau - Interpol'),
     ('Center for', 'Centers for'),
     (' for the District of Columbia', ''),
-    (' - FOIA Program Office (ARSF)', ''),
```
Comment (geramirez) [inline contacts/layer_with_usa_contacts.py:35]: These changes were to simplify the logic bit.

File: contacts/layer_with_usa_contacts.py [hunk 6]
```diff
@@ -109,10 +108,10 @@ def update_dict(old_data, new_data):
     the USA contacts API
     """
 
-    old_data['usa_id'] = new_data['usa_id']
+    old_data['usa_id'] = new_data.get('usa_id')
     if new_data.get('description'):
-        old_data['description'] = new_data['description']
-    if new_data.get('abbreviation'):
+        old_data['description'] = new_data.get('description')
+    if new_data.get('abbreviation') and not old_data.get('abbreviation'):
```
Comment (geramirez) [inline contacts/layer_with_usa_contacts.py:114]: I changed this so that manual data abbreviations get priority.

**Comment by geramirez** [comment]
@khandelwal This should be ready for review once Travis finishes.

**Pull request merged**